packages/kit/types/ambient.d.ts

```diff
@@ -89,24 +89,34 @@ declare module '$app/forms' {
 	import type { ActionResult } from '@sveltejs/kit';
 
 	export type SubmitFunction<
-		Success extends Record<string, unknown> | undefined,
-		Invalid extends Record<string, unknown> | undefined
+		Element extends HTMLFormElement | HTMLInputElement | HTMLButtonElement = HTMLFormElement,
+		Success extends Record<string, unknown> | undefined = Record<string, any>,
+		Invalid extends Record<string, unknown> | undefined = Record<string, any>
 	> = (input: {
 		data: FormData;
 		form: HTMLFormElement;
+		element: Element;
 		cancel: () => void;
-	}) => void | ((result: ActionResult<Success, Invalid>) => void);
+	}) =>
+		| void
+		| ((opts: {
+				data: FormData;
+				form: HTMLFormElement;
+				element: Element;
+				result: ActionResult<Success, Invalid>;
+		  }) => void);
 
 	/**
 	 * This action enhances a `<form>` element that otherwise would work without JavaScript.
 	 * @param form The form element
 	 * @param options Callbacks for different states of the form lifecycle
 	 */
 	export function enhance<
+		Element extends HTMLFormElement | HTMLInputElement | HTMLButtonElement = HTMLFormElement,
 		Success extends Record<string, unknown> | undefined = Record<string, any>,
 		Invalid extends Record<string, unknown> | undefined = Record<string, any>
 	>(
-		form: HTMLFormElement,
+		form: Element,
```

Comment:
doesn't really matter but for the sake of persnickitiness

```suggestion
		element: Element,
```